src/core/Akka/Pattern/CircuitBreaker.cs

```diff
@@ -221,43 +220,99 @@ public CircuitBreaker(IScheduler scheduler, int maxFailures, TimeSpan callTimeou
         /// <summary>
         /// Wraps invocation of asynchronous calls that need to be protected
         /// </summary>
-        /// <typeparam name="T">TBD</typeparam>
+        /// <typeparam name="T">The <see cref="Type"/> returned by the protected function</typeparam>
         /// <param name="body">Call needing protected</param>
         /// <returns><see cref="Task"/> containing the call result</returns>
+        [Obsolete(message:"Use WithCircuitBreaker() that accepts functions with CancellationToken parameter. Since 1.5.42")]
         public Task<T> WithCircuitBreaker<T>(Func<Task<T>> body) => CurrentState.Invoke(body);
 
+        /// <summary>
+        /// Wraps invocation of asynchronous calls that need to be protected
+        /// </summary>
+        /// <typeparam name="T">The <see cref="Type"/> returned by the protected function</typeparam>
+        /// <param name="body">Call needing protected</param>
+        /// <returns><see cref="Task"/> containing the call result</returns>
+        public Task<T> WithCircuitBreaker<T>(Func<CancellationToken, Task<T>> body) => CurrentState.Invoke(body);
```

Comment:
All `.WithCircuitBreaker()` methods now accepts delegate function with a `CancellationToken` as one of its arguments, this `CancellationToken` will be used to pass in cancellation signal downstream into the protected delegate function if the `CircuitBreaker` throws.